src/com/goide/runconfig/testing/frameworks/gocheck/GocheckEventsConverter.java

```diff
@@ -0,0 +1,413 @@
+    }
+    myStdOut = null;
+  }
+
+  @NotNull
+  private static String testUrl(@NotNull String testName) {
+    return GoTestLocationProvider.PROTOCOL + "://" + testName;
+  }
+
+  private void processTestSectionStart(@NotNull String testName, Key outputType, ServiceMessageVisitor visitor) throws ParseException {
+    String suiteName = testName.substring(0, testName.indexOf("."));
+    myTestName = testName;
+    if (!suiteName.equals(mySuiteName)) {
+      if (mySuiteName != null) {
+        String suiteFinishedMsg = ServiceMessageBuilder.testSuiteFinished(mySuiteName).toString();
+        super.processServiceMessages(suiteFinishedMsg, outputType, visitor);
+      }
+      mySuiteName = suiteName;
+      String suiteStartedMsg = ServiceMessageBuilder.testSuiteStarted(suiteName).toString();
+      super.processServiceMessages(suiteStartedMsg, outputType, visitor);
+    }
+  }
+
+  private Map<String, String> parseFailureAttributes() {
+    List<String> buffer = ContainerUtil.newArrayList();
+    Matcher matcher;
+
+    int i = myStdOut.size() - 1;
+    while (StringUtil.isEmptyOrSpaces(myStdOut.get(i))) {
+      --i;
+    }
+
+    String expected = "";
+    while ((matcher = ERROR_CONTINUATION.matcher(myStdOut.get(i))).matches()) {
+      --i;
+      buffer.add(matcher.group(1));
+    }
+    if ((matcher = ERROR_EXPECTED.matcher(myStdOut.get(i))).matches()) {
+      --i;
+      buffer.add(matcher.group(4));
+      Collections.reverse(buffer);
+      expected = StringUtil.join(buffer, "\n");
+      buffer.clear();
+    }
+
+    String actual = "";
+    while ((matcher = ERROR_CONTINUATION.matcher(myStdOut.get(i))).matches()) {
+      --i;
+      buffer.add(matcher.group(1));
+    }
+    if ((matcher = ERROR_ACTUAL.matcher(myStdOut.get(i))).matches()) {
+      --i;
+      buffer.add(matcher.group(4));
+      Collections.reverse(buffer);
+      actual = StringUtil.join(buffer, "\n");
+      buffer.clear();
+    }
+
+    while (!(matcher = ERROR_LOCATION.matcher(myStdOut.get(i))).matches()) {
+      buffer.add(myStdOut.get(i));
+      --i;
+    }
+    Collections.reverse(buffer);
+    String errorMessage = StringUtil.join(buffer, "").trim();
+    String details = matcher.group(1);
+
+    Map<String, String> attributes = ContainerUtil.newHashMap();
+    attributes.put("type", "comparisonFailure");
+    attributes.put("message", errorMessage);
+    attributes.put("details", details);
+    attributes.put("actual", actual);
+    attributes.put("expected", expected);
+
+    // Remove the test results from the standard output and report the rest.
+    myStdOut = myStdOut.subList(0, i);
+
+    return attributes;
+  }
+
+  private Map<String, String> parsePanickedAttributes() {
+    List<String> buffer = ContainerUtil.newArrayList();
+    Matcher matcher;
+
+    // Traverse the test standard output in bottom-up direction and retrieve
+    // the assertion failure parameters.
+
+    int i = myStdOut.size() - 1;
```

Comment:
The same for parseFailureAttributes.

Also any interaction with myStdOut is surprising since it's regularly reassignable and nullable instance-field.